tests/components/devialet/test_media_player.py

```diff
@@ -101,131 +118,140 @@
+        assert state.state == MediaPlayerState.PLAYING
+        assert state.name == NAME
+        assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.2
+        assert state.attributes[ATTR_MEDIA_VOLUME_MUTED] is False
+        assert state.attributes[ATTR_INPUT_SOURCE_LIST] is not None
+        assert state.attributes[ATTR_SOUND_MODE_LIST] is not None
+        assert state.attributes[ATTR_MEDIA_ARTIST] == "The Beatles"
+        assert state.attributes[ATTR_MEDIA_ALBUM_NAME] == "1 (Remastered)"
+        assert state.attributes[ATTR_MEDIA_TITLE] == "Hey Jude - Remastered 2015"
+        assert state.attributes[ATTR_ENTITY_PICTURE] is not None
+        assert state.attributes[ATTR_MEDIA_DURATION] == 425653
+        assert state.attributes[ATTR_MEDIA_POSITION] == 123102
+        assert state.attributes[ATTR_MEDIA_POSITION_UPDATED_AT] is not None
+        assert state.attributes[ATTR_SUPPORTED_FEATURES] is not None
+        assert state.attributes[ATTR_INPUT_SOURCE] is not None
+        assert state.attributes[ATTR_SOUND_MODE] is not None
+
+        with patch.object(
+            DevialetApi,
+            "playing_state",
+            new_callable=PropertyMock,
+        ) as mock:
+            mock.return_value = MediaPlayerState.PAUSED
 
             await hass.config_entries.async_reload(entry.entry_id)
             await hass.async_block_till_done()
             assert (
-                hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}").attributes[
-                    ATTR_SOUND_MODE
-                ]
-                == "Night mode"
+                hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}").state
+                == MediaPlayerState.PAUSED
             )
 
-    with patch.object(DevialetApi, "equalizer", new_callable=PropertyMock) as mock:
-        mock.return_value = "unexpected_value"
-
-        with patch.object(DevialetApi, "night_mode", new_callable=PropertyMock) as mock:
-            mock.return_value = False
+        with patch.object(
+            DevialetApi,
+            "playing_state",
+            new_callable=PropertyMock,
+        ) as mock:
+            mock.return_value = MediaPlayerState.ON
 
             await hass.config_entries.async_reload(entry.entry_id)
             await hass.async_block_till_done()
             assert (
-                ATTR_SOUND_MODE
-                not in hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}").attributes
+                hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}").state
+                == MediaPlayerState.ON
             )
 
-    with patch.object(DevialetApi, "equalizer", new_callable=PropertyMock) as mock:
-        mock.return_value = None
+        with patch.object(DevialetApi, "equalizer", new_callable=PropertyMock) as mock:
+            mock.return_value = None
+
+            with patch.object(
+                DevialetApi, "night_mode", new_callable=PropertyMock
+            ) as mock:
+                mock.return_value = True
+
```

Comment:
Seeing this I would really recommend doing proper mocks, because this isn't ideal. Like, either use library mocks or use http mocks. Don't use both as this becomes hard to maintain

Reply:
I can improve testing in another PR. It requires a lot of time to me and this is already an existing part.

Reply:
Yes but the tests themselves can be a lot cleaner than having multiple nested `with patch`es, you can consider patching the full object or merging the patch statements `with patch(), patch():`

Reply:
So I merged the patch statements at all locations. It's not perfect, but already a lot cleaner I think. Thanks for the suggestion.